src/sys/ptrace.rs

```diff
@@ -1,66 +1,87 @@
+        PTRACE_CONT        = libc::PTRACE_CONT,
+        PTRACE_KILL        = libc::PTRACE_KILL,
+        PTRACE_SINGLESTEP  = libc::PTRACE_SINGLESTEP,
+        PTRACE_GETREGS     = libc::PTRACE_GETREGS,
+        PTRACE_SETREGS     = libc::PTRACE_SETREGS,
+        PTRACE_GETFPREGS   = libc::PTRACE_GETFPREGS,
+        PTRACE_SETFPREGS   = libc::PTRACE_SETFPREGS,
+        PTRACE_ATTACH      = libc::PTRACE_ATTACH,
+        PTRACE_DETACH      = libc::PTRACE_DETACH,
+        PTRACE_GETFPXREGS  = libc::PTRACE_GETFPXREGS,
+        PTRACE_SETFPXREGS  = libc::PTRACE_SETFPXREGS,
+        PTRACE_SYSCALL     = libc::PTRACE_SYSCALL,
+        PTRACE_SETOPTIONS  = libc::PTRACE_SETOPTIONS,
+        PTRACE_GETEVENTMSG = libc::PTRACE_GETEVENTMSG,
+        PTRACE_GETSIGINFO  = libc::PTRACE_GETSIGINFO,
+        PTRACE_SETSIGINFO  = libc::PTRACE_SETSIGINFO,
+        PTRACE_GETREGSET   = libc::PTRACE_GETREGSET,
+        PTRACE_SETREGSET   = libc::PTRACE_SETREGSET,
+        PTRACE_SEIZE       = libc::PTRACE_SEIZE,
+        PTRACE_INTERRUPT   = libc::PTRACE_INTERRUPT,
+        PTRACE_LISTEN      = libc::PTRACE_LISTEN,
+        PTRACE_PEEKSIGINFO = libc::PTRACE_PEEKSIGINFO,
+    }
 
-    pub const PTRACE_TRACEME:     PtraceRequest = 0;
-    pub const PTRACE_PEEKTEXT:    PtraceRequest = 1;
-    pub const PTRACE_PEEKDATA:    PtraceRequest = 2;
-    pub const PTRACE_PEEKUSER:    PtraceRequest = 3;
-    pub const PTRACE_POKETEXT:    PtraceRequest = 4;
-    pub const PTRACE_POKEDATA:    PtraceRequest = 5;
-    pub const PTRACE_POKEUSER:    PtraceRequest = 6;
-    pub const PTRACE_CONT:        PtraceRequest = 7;
-    pub const PTRACE_KILL:        PtraceRequest = 8;
-    pub const PTRACE_SINGLESTEP:  PtraceRequest = 9;
-    pub const PTRACE_GETREGS:     PtraceRequest = 12;
-    pub const PTRACE_SETREGS:     PtraceRequest = 13;
-    pub const PTRACE_GETFPREGS:   PtraceRequest = 14;
-    pub const PTRACE_SETFPREGS:   PtraceRequest = 15;
-    pub const PTRACE_ATTACH:      PtraceRequest = 16;
-    pub const PTRACE_DETACH:      PtraceRequest = 17;
-    pub const PTRACE_GETFPXREGS:  PtraceRequest = 18;
-    pub const PTRACE_SETFPXREGS:  PtraceRequest = 19;
-    pub const PTRACE_SYSCALL:     PtraceRequest = 24;
-    pub const PTRACE_SETOPTIONS:  PtraceRequest = 0x4200;
-    pub const PTRACE_GETEVENTMSG: PtraceRequest = 0x4201;
-    pub const PTRACE_GETSIGINFO:  PtraceRequest = 0x4202;
-    pub const PTRACE_SETSIGINFO:  PtraceRequest = 0x4203;
-    pub const PTRACE_GETREGSET:   PtraceRequest = 0x4204;
-    pub const PTRACE_SETREGSET:   PtraceRequest = 0x4205;
-    pub const PTRACE_SEIZE:       PtraceRequest = 0x4206;
-    pub const PTRACE_INTERRUPT:   PtraceRequest = 0x4207;
-    pub const PTRACE_LISTEN:      PtraceRequest = 0x4208;
-    pub const PTRACE_PEEKSIGINFO: PtraceRequest = 0x4209;
+    // These aren't currently in libc.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    #[derive(Eq, PartialEq, Clone, Copy, Debug)]
+    #[repr(u32)]
+    pub enum PtraceEvent {
```

Comment:
I think we should use `#[repr(i32)]` and thus consistently the type, which is used by the ffi function.